tests/unit/test_serialization.py

```diff
@@ -755,3 +770,42 @@ def test_serialize_deserialize_double_round_trip(deserialize, serialize, seriali
 
             np.testing.assert_array_equal(nan_a, nan_b)
             np.testing.assert_allclose(field_result_a[~nan_a], field_result_b[~nan_b], rtol=1e-15)
+
+
+@pytest.mark.parametrize(
+    ("deserialize", "pack", "data_filter"),
+    (
+        pytest.param(json_deserialize, to_json, ComponentAttributeFilterOptions.ALL, id="json.all"),
+        pytest.param(json_deserialize, to_json, ComponentAttributeFilterOptions.RELEVANT, id="json.relevant"),
+        pytest.param(msgpack_deserialize, to_msgpack, ComponentAttributeFilterOptions.ALL, id="msgpack.all"),
+        pytest.param(msgpack_deserialize, to_msgpack, ComponentAttributeFilterOptions.RELEVANT, id="msgpack.relevant"),
+        pytest.param(
+            json_deserialize,
+            to_json,
+            {
+                "node": ["id"],
```

Comment:
```suggestion
                "node": None,
```
this is also an option that should return row-based for that component. Not sure if you test this as is

Reply:
Yup, works as intended. Added in the coming commit.